slack_sdk/web/async_client.py

```diff
@@ -2992,6 +2995,160 @@ async def files_upload(
             kwargs["content"] = content
             return await self.api_call("files.upload", data=kwargs)
 
+    async def files_upload_v2(
+        self,
+        *,
+        # for sending a single file
+        filename: Optional[str] = None,  # you can skip this only when sending along with content parameter
+        file: Optional[Union[str, bytes, IOBase]] = None,
+        content: Optional[Union[str, bytes]] = None,
+        title: Optional[str] = None,
+        alt_txt: Optional[str] = None,
+        snippet_type: Optional[str] = None,
+        filetype: Optional[str] = None,  # no longer supported
+        # To upload multiple files at a time
+        upload_files: Optional[List[Dict[str, Any]]] = None,
+        channel: Optional[str] = None,
+        channels: Optional[Union[str, Sequence[str]]] = None,  # having n channels is no longer supported
```

Comment:
I intentionally didn't remove this keyword argument. Unlike the Java SDK, there is no compiler detection in Python. Thus, removing the argument name that existed in files.upload (v1) can result in many developers' confusion. 

That being said, sharing files in multiple channels is no longer supported. So, I've added runtime validation below to   inform the breaking change.

Reply:
@WilliamBergamin Thanks. I think it's fine to move this argument to kwargs, but I still would like to support it as long as the list has only one item. I am pretty sure that most of the existing code has only a single channel in this parameter. Perhaps, we can add a warning logging even for the single item use case, but we can allow it for smoother migration.